python/ray/util/placement_group.py

```diff
@@ -222,32 +214,8 @@ def placement_group(
             f"Invalid hex ID of _soft_target_node_id, got {_soft_target_node_id}"
         )
 
-    # Validate bundles
-    for bundle in bundles:
-        if len(bundle) == 0 or all(
-            resource_value == 0 for resource_value in bundle.values()
-        ):
-            raise ValueError(
-                "Bundles cannot be an empty dictionary or "
-                f"resources with only 0 values. Bundles: {bundles}"
-            )
-
-        if "object_store_memory" in bundle.keys():
-            warnings.warn(
-                "Setting 'object_store_memory' for"
-                " bundles is deprecated since it doesn't actually"
-                " reserve the required object store memory."
-                f" Use object spilling that's enabled by default (https://docs.ray.io/en/{get_ray_doc_version()}/ray-core/objects/object-spilling.html) "  # noqa: E501
-                "instead to bypass the object store memory size limitation.",
-                DeprecationWarning,
-                stacklevel=1,
-            )
-
-    if strategy not in VALID_PLACEMENT_GROUP_STRATEGIES:
-        raise ValueError(
-            f"Invalid placement group strategy {strategy}. "
-            f"Supported strategies are: {VALID_PLACEMENT_GROUP_STRATEGIES}."
-        )
+    validate_bundles(bundles)
+    validate_strategy(strategy)
```

Comment:
I slightly prefer to adding a `_dry_run` option to the existing `placement_group()` function.

Reply:
Why is that? I figured pulling out the validation logic would make it easier to centralize, test, and reuse that logic.

Reply:
I feel at least we should have a single `validate_placement_group()` function that takes the same parameters as `placement_group()`. Otherwise, we need to have a `validate_xxx()` for each parameter (this is limiting since sometimes the validation is for the the combination of multiple parameters: e.g. mutual exclusiveness of two parameters)

By having a `_dry_run` option, we don't need to have a separate validate function and keep the interface in-sync with the real `placement_group()`.